src/rp2_common/hardware_clocks/include/hardware/clocks.h

```diff
@@ -318,6 +318,31 @@ typedef clock_num_t clock_handle_t;
  */
 bool clock_configure(clock_handle_t clock, uint32_t src, uint32_t auxsrc, uint32_t src_freq, uint32_t freq);
 
+/*! \brief Configure the specified clock with 1MHz accuracy
+ *  \ingroup hardware_clocks
+ *
+ * This function differs from clock_configure in that it does not configure the clocks as accurately,
+ * but therefore doesn't need to bring in 64-bit division functions, reducing the code size if 64-bit
+ * division is not otherwise used by the application.
+ *
+ * \if rp2350_specific
+ * Note: The RP2350 clock hardware supports divisors from 1.0->65536.0 in steps of 1/65536
+ *
+ * \endif
+ * \if rp2040_specific
+ * Note: The RP2040 clock hardware only supports divisors of exactly 1.0 or 2.0->16777216.0 in steps of 1/256
+ * \endif
+ * 
+ * See the tables in the description for details on the possible values for clock sources.
+ *
+ * \param clock The clock to configure
+ * \param src The main clock source, can be 0.
+ * \param auxsrc The auxiliary clock source, which depends on which clock is being set. Can be 0
+ * \param src_freq_mhz Frequency of the input clock source in MHz
+ * \param freq_mhz Requested frequency in MHz
```

Comment:
Looks like the doxygen is missing a `\return` ?

Reply:
```suggestion
 * \param freq_mhz Requested frequency in MHz
 * \return true if the clock is updated, false if freq > src_freq
```